server/src/diagnostics.ts

```diff
@@ -191,6 +191,37 @@ export async function perlcritic(textDocument: TextDocument, workspaceFolders: W
     return diagnostics;
 }
 
+export async function perlimports(textDocument: TextDocument, workspaceFolders: WorkspaceFolder[] | null, settings: NavigatorSettings): Promise<Diagnostic[]> {
+    if(!settings.perlimportsLintEnabled) return [];
+    const importsPath = join(getPerlAssetsPath(), 'perlimportsWrapper.pl');
+    const cliParams = [importsPath, ...getPerlimportsProfile(settings), '--lint', '--json', '--filename', Uri.parse(textDocument.uri).fsPath];
+
+    nLog("Now starting perlimports with: " + cliParams.join(" "), settings);
+
+    const diagnostics: Diagnostic[] = [];
+    let output: string;
+    try {
+        const process = async_execFile(settings.perlPath, cliParams, {timeout: 25000});
+        process?.child?.stdin?.on('error', (error: any) => {
+            nLog("perlimports Error Caught: " + error, settings);
+        });
+        process?.child?.stdin?.end();
+        const out = await process;
```

Comment:
The `code` is not actually passed to perlimports for linting, so it ends up linting the saved version of the file instead of reading from stdin. This seems like a bug in perlimports. Given that --read-stdin is passed, shouldn't perl imports throw an errors instead of reading from --filename?

Reply:
I will take a closer look at this. `perlimports`, on the tidying side, accepts input on `STDIN`, but still needs the filename so that it can eval the entire package and use that to tidy the input. That leaves a mismatch when you have an unsaved buffer, so in `vim` I've been getting around that by saving the document before running `perlimports`.

Having said that, what `perlimports` really should be able to do is accept an entire document on `STDIN` as well as a parameter which defines which range of lines should be checked, so that we don't have to rely on files being saved first.

The linting part is still fairly new, so there may be a bug there.

I really just need to fix `perlimports` not to read files in this case, though. I'll move it higher up my TODO list.